src-self-hosted/libc_installation.zig

```diff
@@ -34,6 +38,50 @@ pub const LibCInstallation = struct {
         ZigIsTheCCompiler,
     };
 
+    fn envSubstitution(allocator: *Allocator, input: [:0]const u8, stderr: anytype) ![:0]u8 {
+        var result = try ArrayList(u8).initCapacity(allocator, input.len + 1);
+        errdefer result.deinit();
+
+        var i: usize = 0;
+        outer: while (i < input.len) : (i += 1) {
+            if (input[i] == '$') {
+                for (input[i + 1..]) |c, j| {
+                    const input_offset = i + j + 1;
+                    if ((!ascii.isAlNum(c) and c != '_') or input_offset == input.len - 1) {
+                        const needle = if (input_offset == input.len - 1)
+                            input[i..input_offset + 1]
+                        else
+                            input[i..input_offset];
+
+                        const name = needle[1..];
+
+                        const value = process.getEnvVarOwned(allocator, name) catch |err| switch (err) {
+                            error.EnvironmentVariableNotFound => {
+                                try stderr.print("environment variable ${} specified but not found\n", .{name});
+                                return error.ParseError;
+                            },
+                            error.InvalidUtf8 => {
+                                try stderr.print("environment variable ${} contains invalid utf-8\n", .{name});
+                                return error.ParseError;
+                            },
+                            error.OutOfMemory => |e| return e,
+                        };
+
+                        try result.appendSlice(value);
+                        i += needle.len;
+                        continue :outer;
```

Comment:
why continue here? (it should automatically)

Reply:
We want to exit the `for` loop, since we have now hit the end of the environment variable name.

Reply:
oh; doh. in that case just a `break` would work.